> > > No worries. Incompatible change from Spark: https://issues.apache.org/jira/browse/SPARK-41970 I have created a branch for Spark-3.2 Let's support spark-3.4 on the master branch. This PR should only support Spark 3.4. Please address the comments above.
> > 
> > 
> > @junshi15 it seems that the current master also supports spark-3.3, do you think we should create a branch for spark-3.3?
> 
> Does it work out of box or you have to change pom.xml and recompile? I have not tested either of them. If you can verify the binary works for both 3.2 and 3.3, then we can just rename the branch as spark-3.2-3.3. If you have to change pom.xml, then let's create a separate branch and spin a binary. My guess is the former, but please verify it if you have time.

@junshi15 I think the pom.xml is the default configuration, which can be changed by the passed parameters to `mvn` command. I've verified that current master branch can pass all unit tests for spark-3.2 and spark-3.3, however, spark-3.4 will fail with the error @ssiegel95 fixed in this PR. 
- spark-3.2 passed: `mvn -Pscala-2.13 -Dspark.version=3.2.0 test`:
```
...
Run completed in 9 seconds, 690 milliseconds.
Total number of tests run: 31
Suites: completed 6, aborted 0
Tests: succeeded 31, failed 0, canceled 0, ignored 0, pending 0
All tests passed.
```
- spark-3.3 passed: `mvn -Pscala-2.13 -Dspark.version=3.3.0 test`:
```
Run completed in 10 seconds, 397 milliseconds.
Total number of tests run: 31
Suites: completed 6, aborted 0
Tests: succeeded 31, failed 0, canceled 0, ignored 0, pending 0
All tests passed.
```
- spark-3.4 failed: `mvn -Pscala-2.13 -Dspark.version=3.4.0 test`:
```
[ERROR] /home/mizhou/tmp/0611/spark-tfrecord/src/main/scala/com/linkedin/spark/datasources/tfrecord/TFRecordFileReader.scala:26: type mismatch;
 found   : org.apache.spark.paths.SparkPath
 required: String
[ERROR]       new Path(new URI(file.filePath)),
[ERROR]                             ^
[ERROR] one error found
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 7.222 s
[INFO] Finished at: 2023-06-11T22:56:05-07:00
[INFO] Final Memory: 94M/1444M
```

--- a/README.md
+++ b/README.md
@@ -12,6 +12,8 @@ The artifacts are published to [bintray](https://bintray.com/linkedin/maven/spar
 - Version 0.3.x targets Spark 3.0 and Scala 2.12
 - Version 0.4.x targets Spark 3.2 and Scala 2.12
 - Version 0.5.x targets Spark 3.2 and Scala 2.13
+- Version 0.6.x targets Spark 3.4 and Scala 2.12
+- Version 0.7.x targets Spark 3.4 and Scala 2.13
 
 To use the package, please include the dependency as follows
 

--- a/src/main/scala/com/linkedin/spark/datasources/tfrecord/TFRecordFileReader.scala
+++ b/src/main/scala/com/linkedin/spark/datasources/tfrecord/TFRecordFileReader.scala
@@ -1,7 +1,7 @@
 package com.linkedin.spark.datasources.tfrecord
 
-import java.net.URI
 import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.Path
 import org.apache.hadoop.mapreduce.{JobID, TaskAttemptID, TaskID, TaskType}
 import org.apache.hadoop.mapreduce.lib.input.FileSplit
 import org.apache.hadoop.mapreduce.task.TaskAttemptContextImpl

